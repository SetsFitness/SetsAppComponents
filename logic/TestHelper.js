import ably from "../redux/reducers/ablyReducer";
import auth from "../redux/reducers/authReducer";
import cache from "../redux/reducers/cacheReducer";
import info from "../redux/reducers/infoReducer";
import message from "../redux/reducers/messageReducer";
import search from "../redux/reducers/searchReducer";
import user from "../redux/reducers/userReducer";

/**
 * Class used for testing certain parts of the code that may try to connect to AWS in order to complete. Run this when
 * we are automatically running a test.
 */
export default class TestHelper {
    static ifTesting = false;
    static setTest() { TestHelper.ifTesting = true; }
    static unsetTest() { TestHelper.ifTesting = false; }
}

/**
 * Gets the entire redux store components given for initializing a mock store. Allows you to only initialize what you
 * need to for the test.
 *
 * TODO Edit when we use more reducers.
 *
 * @param {[string]} storeTypes The strings for the types of the redux store to initialize.
 * @return {{ably: *, auth: *, cache: *, info: *, message: *, user: *, search: *}}
 */
export const getInitalReduxStore = (storeTypes) => {
    return {
        ably: !storeTypes || storeTypes.includes("ably") ? ably(undefined, {type: "__NOT_A_REAL_ACTION__"}) : null,
        auth: !storeTypes || storeTypes.includes("auth") ? auth(undefined, {type: "__NOT_A_REAL_ACTION__"}) : null,
        cache: !storeTypes || storeTypes.includes("cache") ? cache(undefined, {type: "__NOT_A_REAL_ACTION__"}) : null,
        info: !storeTypes || storeTypes.includes("info") ? info(undefined, {type: "__NOT_A_REAL_ACTION__"}) : null,
        message: !storeTypes || storeTypes.includes("message") ? message(undefined, {type: "__NOT_A_REAL_ACTION__"}) : null,
        user: !storeTypes || storeTypes.includes("user") ? user(undefined, {type: "__NOT_A_REAL_ACTION__"}) : null,
        search: !storeTypes || storeTypes.includes("search") ? search(undefined, {type: "__NOT_A_REAL_ACTION__"}) : null,
    }
};