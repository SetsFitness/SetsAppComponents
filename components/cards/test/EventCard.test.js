import React from 'react';
import EventCard from "../EventCard";
import { shallow, mount } from 'enzyme';
import "../../../testing/SetTesting";
import {store} from "../../../testing/TestHelper";
import {expect} from "chai";

const event1 = {
    id: "EV0001",
    title: "Sleeny's event",
    time: "2019-09-15T15:53:00",
    time_created: "2019-07-14T15:53:00",
    capacity: 5,
    members: ['Blake', 'Kaloob', 'Leo']
};

it('renders without crashing', () => {
    const component = shallow(<EventCard store={store()} event={event1}/>);
    expect(component).toMatchSnapshot();
});

describe("Event Card", () => {
    // States ~
    describe("States", () => {
        // Here you check the state of the component, like the shown text and sub-components for the component.
        // Make sure you go into all the conditions of the component so that you test every possible state of the
        // component.
        // EXAMPLE: it("Shows the spinner when no post is passed in");
        it('Event card without event prop displays spinner', () => {
            const component = shallow(<EventCard store={store()} event={null}/>);

            expect(component.find("Spinner")).to.have.lengthOf(1);
        });

        it('Event card with event prop displays card', () => {
            const component = shallow(<EventCard store={store()} event={event1}/>);

            expect(component.find("Card")).to.have.lengthOf(1);
        });

        it('Event card with event prop displays event title in header', () => {
            const component = shallow(<EventCard store={store()} event={event1}/>);

            console.log(component.find("CardHeader").render().text());

            expect(component.find("CardHeader").render().text()).equals("Sleeny's event");
        });

        it('Event card with trainer prop displays event date properly', () => {
            const component = shallow(<EventCard store={store()} event={event1}/>);

            console.log(component.find("CardMeta").at(0).render().text());

            expect(component.find("CardMeta").at(0).render().text()).equals("Sun, Sep 15, 2019 03:53PM");
        });

        it('Event card with event prop displays event ', () => {
            const component = shallow(<EventCard store={store()} event={event1}/>);

            console.log(component.find("CardMeta").at(1).render().text());

            expect(component.find("CardMeta").at(1).render().text()).equals("Sun, Jul 14, 2019 03:53PM");
        });
    });
    // Events ~
    describe("Events", () => {
        // Here you use .simultate('click') to check the change in the component.
        // Only in special circumstances should you use jest.fn() and that's if you're directly passing in a function
        // into the component's props and you have access to it.

        //TODO: Test that pressing the card sets descriptionModalOpen to true
        it('trainer card with trainer prop displays trainer name in header', () => {
            const component = shallow(<EventCard store={store()} event={event1}/>);

            console.log(component.state());

            expect(component.find("CardHeader").render().text()).equals("Sleeny");
        });
    });
});
