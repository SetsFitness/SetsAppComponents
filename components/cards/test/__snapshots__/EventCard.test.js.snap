// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventCard
    event={
      Object {
        "capacity": 5,
        "id": "EV0001",
        "members": Array [
          "Blake",
          "Kaloob",
          "Leo",
        ],
        "time": "2019-09-15T15:53:00",
        "time_created": "2019-07-14T15:53:00",
        "title": "Sleeny's event",
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <CardContent>
          <CardHeader
            textAlign="center"
          >
            Sleeny's event
          </CardHeader>
          <CardMeta
            textAlign="center"
          >
            Sun, Sep 15, 2019 03:53PM
          </CardMeta>
          <Connect(EventDescriptionModal)
            eventID="EV0001"
            onClose={[Function]}
            open={false}
          />
        </CardContent>,
        <CardContent
          extra={true}
        >
          <CardMeta>
            Sun, Jul 14, 2019 03:53PM
          </CardMeta>
          <CardMeta
            textAlign="center"
          >
            3
             of 
            5
             spots taken.
          </CardMeta>
        </CardContent>,
      ],
      "fluid": true,
      "onClick": [Function],
      "raised": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <CardHeader
              textAlign="center"
            >
              Sleeny's event
            </CardHeader>,
            <CardMeta
              textAlign="center"
            >
              Sun, Sep 15, 2019 03:53PM
            </CardMeta>,
            <Connect(EventDescriptionModal)
              eventID="EV0001"
              onClose={[Function]}
              open={false}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Sleeny's event",
              "textAlign": "center",
            },
            "ref": null,
            "rendered": "Sleeny's event",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Sun, Sep 15, 2019 03:53PM",
              "textAlign": "center",
            },
            "ref": null,
            "rendered": "Sun, Sep 15, 2019 03:53PM",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "eventID": "EV0001",
              "onClose": [Function],
              "open": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <CardMeta>
              Sun, Jul 14, 2019 03:53PM
            </CardMeta>,
            <CardMeta
              textAlign="center"
            >
              3
               of 
              5
               spots taken.
            </CardMeta>,
          ],
          "extra": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Sun, Jul 14, 2019 03:53PM",
            },
            "ref": null,
            "rendered": "Sun, Jul 14, 2019 03:53PM",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                3,
                " of ",
                5,
                " spots taken.",
              ],
              "textAlign": "center",
            },
            "ref": null,
            "rendered": Array [
              3,
              " of ",
              5,
              " spots taken.",
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <CardContent>
            <CardHeader
              textAlign="center"
            >
              Sleeny's event
            </CardHeader>
            <CardMeta
              textAlign="center"
            >
              Sun, Sep 15, 2019 03:53PM
            </CardMeta>
            <Connect(EventDescriptionModal)
              eventID="EV0001"
              onClose={[Function]}
              open={false}
            />
          </CardContent>,
          <CardContent
            extra={true}
          >
            <CardMeta>
              Sun, Jul 14, 2019 03:53PM
            </CardMeta>
            <CardMeta
              textAlign="center"
            >
              3
               of 
              5
               spots taken.
            </CardMeta>
          </CardContent>,
        ],
        "fluid": true,
        "onClick": [Function],
        "raised": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <CardHeader
                textAlign="center"
              >
                Sleeny's event
              </CardHeader>,
              <CardMeta
                textAlign="center"
              >
                Sun, Sep 15, 2019 03:53PM
              </CardMeta>,
              <Connect(EventDescriptionModal)
                eventID="EV0001"
                onClose={[Function]}
                open={false}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Sleeny's event",
                "textAlign": "center",
              },
              "ref": null,
              "rendered": "Sleeny's event",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Sun, Sep 15, 2019 03:53PM",
                "textAlign": "center",
              },
              "ref": null,
              "rendered": "Sun, Sep 15, 2019 03:53PM",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "eventID": "EV0001",
                "onClose": [Function],
                "open": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <CardMeta>
                Sun, Jul 14, 2019 03:53PM
              </CardMeta>,
              <CardMeta
                textAlign="center"
              >
                3
                 of 
                5
                 spots taken.
              </CardMeta>,
            ],
            "extra": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Sun, Jul 14, 2019 03:53PM",
              },
              "ref": null,
              "rendered": "Sun, Jul 14, 2019 03:53PM",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  3,
                  " of ",
                  5,
                  " spots taken.",
                ],
                "textAlign": "center",
              },
              "ref": null,
              "rendered": Array [
                3,
                " of ",
                5,
                " spots taken.",
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
